vms/platformvm/txs/staker_tx.go

```diff
@@ -42,15 +42,24 @@ type PermissionlessStaker interface {
 	Stake() []*avax.TransferableOutput
 }
 
-type Staker interface {
+type commonStaker interface {
 	SubnetID() ids.ID
 	NodeID() ids.NodeID
 	// PublicKey returns the BLS public key registered by this transaction. If
 	// there was no key registered by this transaction, it will return false.
 	PublicKey() (*bls.PublicKey, bool, error)
-	StartTime() time.Time
-	EndTime() time.Time
 	Weight() uint64
-	PendingPriority() Priority
 	CurrentPriority() Priority
 }
+
+type Staker interface {
+	commonStaker
+	Duration() time.Duration
+}
+
+type PreDForkStaker interface {
```

Comment:
Why do we need a new type here? Couldn't we add `Duration` to the `Staker` interface?

Reply:
It's much cleaner. Once the fork activates, start and end times won't make sense anymore and it just right that we won't methods to access them individually in the `StakerTx` interface.
I think it reduced code complexity. Also the places in code where we have pre fork txs are clearly found via the type

Reply:
Why wouldn't `StartTime` and `EndTime` make sense? Won't we still want to keep track of when the current staking period has started and ended?

Reply:
These are tx interface. StartTime won't be specified in the stakers tx, but will be the chain time when the tx is accepted. So we still have a StartTime on the staker object, but we should not have it in the tx.